pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+
+import argparse
+import os
+
+# THIRD-PARTY MODULES
+
+import numpy as np
+import astropy.units as u
+from astropy.table import Table
+from astropy.coordinates.angle_utilities import angular_separation
+from gammapy.spectrum import cosmic_ray_flux, CrabSpectrum  # UPDATE TO LATEST
+
+# THIS PACKAGE
+
+from pyirf.io.io import load_config, read_FITS
+from pyirf.perf import (
+    CutsOptimisation,
+    CutsDiagnostic,
+    CutsApplicator,
+    IrfMaker,
+    SensitivityMaker,
+)
+
+
+def main():
+
+    # =========================================================================
+    #                   READ INPUT FROM CLI AND CONFIGURATION FILE
+    # =========================================================================
+
+    # INPUT FROM CLI
+
+    parser = argparse.ArgumentParser(description="Produce DL3 data from DL2.")
+
+    parser.add_argument(
+        "--config_file",
+        type=str,
+        required=True,
+        help="A configuration file like pyirf/resources/performance.yaml .",
+    )
+
+    parser.add_argument(
+        "--obs_time",
+        type=str,
+        required=True,
+        help="An observation time written as (value.unit), e.g. '50.h'",
+    )
+
+    parser.add_argument(
+        "--pipeline",
+        type=str,
+        required=True,
+        help="Name of the pipeline that has produced the DL2 files.",
+    )
+
+    parser.add_argument(
+        "--debug", action="store_true", help="Print debugging information."
+    )
+
+    args = parser.parse_args()
+
+    # INPUT FROM THE CONFIGURATION FILE
+
+    cfg = load_config(args.config_file)
+
+    # Add obs. time to the configuration file
+    str_obs_time = args.obs_time.split(".")
+    cfg["analysis"]["obs_time"] = {
+        "value": float(str_obs_time[0]),
+        "unit": str(str_obs_time[-1]),
+    }
+
+    # Get input directory
+    indir = cfg["general"]["indir"]
+
+    # Get template of the input file(s)
+    template_input_file = cfg["general"]["template_input_file"]
+
+    # Get output directory
+    outdir = os.path.join(
+        cfg["general"]["outdir"],
+        "irf_{}_Time{}{}".format(  # name could change in future
+            args.pipeline,
+            # cfg["analysis"]["thsq_opt"]["type"], # not needed now
+            # cfg["analysis"]["obs_time"]["value"],
+            # cfg["analysis"]["obs_time"]["unit"],
+            args.obs_time.split(".")[0],
+            args.obs_time.split(".")[-1],
+        ),
+    )  # and create it if necessary
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+
+    # =========================================================================
+    #                   READ DL2 DATA AND STORE IT ACCORDING TO GADF
+    # =========================================================================
+
+    # Load FITS data
+    particles = ["gamma", "electron", "proton"]
+    evt_dict = dict()  # Contain DL2 file for each type of particle
+    for particle in particles:
+        if args.debug:
```

Comment:
See #22

Reply:
I would do this in a dedicated PR, there is enough material in this one I think (as it is mainly needed to give a runnable script)

Reply:
also because you already opened an issue about this, so we'll not forget